### ISSUE TITLE
View & understand recurring donations being sent to a Project

### ISSUE THREAD

**Issue by divine-comedian** [Giveth/giveth-dapps-v2#3093]
We'll need to show on the project page a list of the recurring donations currently being made to the project.

This information needs to be included on the table:

- [x] date the recurring donation was started
- [x] donor name
- [x] network 
- [x] amount being donated monthly
- [x] total donated to this project from this donor from recurring donation
- [x] USD value of total donated amount

On the scroll bar above the table we should add in a new button - Recurring Donations - clicking this will change the table view to show only the recurring donations.

![Project Single - Donations](https://github.com/Giveth/giveth-dapps-v2/assets/67759413/eeb13fc3-170e-4fe1-b590-497938fc1039)

**Comment by jainkrati** [comment]
@MohammadPCh @mohammadranjbarz can you share date of completion for this from FE and BE perspective.

**Comment by MohammadPCh** [comment]
@jainkrati, the front-end requires approximately ~16 hours to implement this.

**Comment by divine-comedian** [comment]
Thanks @MohammadPCh - Good question on the edge cases, I think a few things will need to be considered. 

## $ USD value of total recurring donations

First is this part -
![image](https://github.com/Giveth/giveth-dapps-v2/assets/67759413/dc51be6d-6694-43e4-8507-7f77a58e7cb2)

This is the dollar value sum of the active recurring donations going to this project for every token (and at a later point, for every chain). Do you think we can do it?

On the QF donation tabs we should not add this text, it only applies to the all time donation & recurring donation tabs for the project page.

## Inactive donations

Other thing in the table itself:
![image](https://github.com/Giveth/giveth-dapps-v2/assets/67759413/c9607f99-3078-468d-96b5-45a80bdc366a)

We should handle when a recurring donation is terminated by the user or ends on its own. Since technically it is no longer streaming funds we should reflect that on the table. The simplest change we can change the amount to "Finalized"

![image](https://github.com/Giveth/giveth-dapps-v2/assets/67759413/beb509e1-d5a5-4946-96ff-e6a919cc70fd)

Also when the donation is no longer active will we have issues querying things like the total amount of tokens sent? And also the dollar value of that total amount. 

I updated https://github.com/Giveth/impact-graph/issues/1142 and also made https://github.com/Giveth/impact-graph/issues/1243



## tx link 

on the total donated column there is a link - while it is a bit complicated to show all of the recurring donation data we can show the  'createFlow' tx - or whatever batch tx that included the createFlow function call. 

https://mumbai.polygonscan.com/tx/0x7845ac65b07ece9a4d2da0c94d1fe3032e62a0ce840f77e4b4c97f863ddbd056

**Comment by divine-comedian** [comment]
@MohammadPCh - what is the latest update for this issue? are you blocked by the back-end or can we proceed?

**Comment by MohammadPCh** [comment]
@divine-comedian  Yes, I am currently waiting for the backend.

**Comment by maryjaf** [comment]
when the below bug  be resolved let me know to start testing 
<img width="1190" alt="image" src="https://github.com/Giveth/giveth-dapps-v2/assets/111529185/5ed1cc27-1603-46f1-aac5-185148d11898">

**Comment by maryjaf** [comment]
Now i can see the recurring donations of project but there are some problems 
- user name isn't shown 
- Total Donated and USD Value is shown 0 
<img width="975" alt="image" src="https://github.com/Giveth/giveth-dapps-v2/assets/111529185/ea1d770a-7ba4-4e02-9efb-e873d866095a">
cc: @alireza-sharifpour

**Comment by alireza-sharifpour** [comment]
> Now i can see the recurring donations of project but there are some problems
> 
> * user name isn't shown
> * Total Donated and USD Value is shown 0
> 
> <img alt="image" width="975" src="https://private-user-images.githubusercontent.com/111529185/313455480-ea1d770a-7ba4-4e02-9efb-e873d866095a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA2NjQ4MDgsIm5iZiI6MTcxMDY2NDUwOCwicGF0aCI6Ii8xMTE1MjkxODUvMzEzNDU1NDgwLWVhMWQ3NzBhLTdiYTQtNGUwMi05ZWZiLWU4NzNkODY2MDk1YS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMxN1QwODM1MDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01ZTk2NjcxNDdmZTc2NjlmODU5NDdhZmU3MzRmNjllYWNhZTYzMjkxMDI3Mjc4YTk2ODdmNzE4MTkwYWQwZmM5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.5CLwKN3w3dMj4K2oh2PDTiMQqfvzkYbqj7md3lVaa2g"> cc: @alireza-sharifpour

About showing the name I'll check it,
Showing 0 is related to backend.

**Comment by maryjaf** [comment]
> About showing the name I'll check it,
> Showing 0 is related to backend.

@mohammadranjbarz Please take a look

**Comment by alireza-sharifpour** [comment]
Is this issue fixed @CarlosQ96?

**Comment by maryjaf** [comment]
Problem of not showing recurring donations of project in donation tab has been resolved 
but this issue would be closed when problem of total donated and usd amount be fixed 

<img width="928" alt="image" src="https://github.com/Giveth/giveth-dapps-v2/assets/111529185/4fb9edb3-9eab-4f93-b899-966a4258d4c0">

**Comment by maryjaf** [comment]
> Problem of not showing recurring donations of project in donation tab has been resolved but this issue would be closed when problem of total donated and usd amount be fixed
> 
> <img alt="image" width="928" src="https://private-user-images.githubusercontent.com/111529185/318301477-4fb9edb3-9eab-4f93-b899-966a4258d4c0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTIwNTI5OTMsIm5iZiI6MTcxMjA1MjY5MywicGF0aCI6Ii8xMTE1MjkxODUvMzE4MzAxNDc3LTRmYjllZGIzLTllYWItNGY5My1iODk5LTk2NmE0MjU4ZDRjMC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDAyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQwMlQxMDExMzNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04MzE4YmRiZWUxYjE2Y2UzNzc3NDRkNDVlYTBkNWFmMDk1YTg1MDk1ZThlZGMyM2MzN2IyODZmY2Y5YmI0MzhmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9._9rFvgCbppeSlUDAAJrHvGwfQlAxsHMfS-oqCynkp0g">

There is a separate issue for this and we could call this issue as Done
and track the status of this problem on this issue 
https://github.com/Giveth/impact-graph/issues/1437

**Comment by mohammadranjbarz** [comment]
@MohammadPCh  @alireza-sharifpour 
I think we have all needed webservices that Mitch has mentioned in the issue, if there is anything that we dint have it in the backend, please let us know

FYI @jainkrati

**Comment by maryjaf** [comment]
> There is a separate issue for this and we could call this issue as Done
> and track the status of this problem on this issue
> [Giveth/impact-graph#1437](https://github.com/Giveth/impact-graph/issues/1437)

@mohammadranjbarz I think you haven't seen this comment 
I've moved it to done and for last problem we have a separate  issue (# 1437)

**Comment by maryjaf** [comment]
Now only active donation is shown in all counting for example if 22 donation is shown for a user, after ending this number decrease to 21
We thought active recurring donation should be counted

**Comment by MohammadPCh** [comment]
@maryjaf Yeah, we need to change our counting policy and count ended donations too.

**Comment by MohammadPCh** [comment]
@divine-comedian, what about including ended donations in the user's public profile? I think it's necessary.

**Comment by maryjaf** [comment]
> based on our previous expectations that I found it was wrong, only active donation is shown in number of user donations, for example if 22 donation is shown for a user, after ending this number decreased to 21 We thought only active recurring donation should be counted every where based on above comment, should this logic be changed also in user page or not ? @divine-comedian

showing "donation finalized" for ended donation in project donation table has been done by @MohammadPCh 

![image](https://github.com/Giveth/giveth-dapps-v2/assets/111529185/c77c45cb-7443-4cdc-b312-1436ca404d67)


**only this problem (number of all donation in user's page and project donation tab) remains that if we want to change logic, it should be fixed by backend**

**Comment by MohammadPCh** [comment]
@divine-comedian for sure, I've fixed the project page. I can fix the user page too.

**Comment by MohammadPCh** [comment]
@divine-comedian we need BE to fix the donations count in all views too. 
cc: @jainkrati @mohammadranjbarz

**Comment by maryjaf** [comment]
Since the ended donation should be shown in table of project donation and user donation, the archived donation is also should be shown, am I right? @MohammadPCh @divine-comedian 

and these field should be changed based on considering ended/archived donations
1- Total number of project donations
2- Total number of user donations
3- Number of contributor 
cc: @mohammadranjbarz

**Comment by MohammadPCh** [comment]
@maryjaf also we need to include the ended amounts in totalAmount.

**Comment by maryjaf** [comment]
Total amount is correct and the ended donations is included in sum of donations 
![image](https://github.com/Giveth/giveth-dapps-v2/assets/111529185/190d4dd7-f18f-4239-ad70-38f0c171005e)

**Comment by divine-comedian** [comment]
Archived and Ended should both be inlcuded in the sum & count

**Comment by mohammadranjbarz** [comment]
> Archived and Ended should both be inlcuded in the sum & count

@maryjaf Can you test it?

**Comment by maryjaf** [comment]
> > Archived and Ended should both be inlcuded in the sum & count
> 
> @maryjaf Can you test it?

It seems it has not been resolved 
take a look on this user, number of one time donations and recurring donations(active, ended, archive) is more than 20 
https://staging.giveth.io/user/0xa1179f64638adb613ddaac32d918eb6beb824104?tab=donations
@mohammadranjbarz

**Comment by maryjaf** [comment]
> > Archived and Ended should both be inlcuded in the sum & count
> 
> @maryjaf Can you test it?

It seems my comment that I had left on thursday,  has been removed by mistake 
I've tested this issue and there is no change and the ended/archived donations are not counted  
@mohammadranjbarz

**Comment by maryjaf** [comment]
> @divine-comedian we need BE to fix the donations count in all views too. cc: @jainkrati @mohammadranjbarz

This issue  is assigned to @mohammadranjbarz, since he won't be available, this issue should be assigned to another one
@jainkrati

**Comment by jainkrati** [comment]
@CarlosQ96 pls guide @RamRamez to fix this issue, if not already fixed by mohammad

**Comment by RamRamez** [comment]
@maryjaf to test and verify.
I fixed it and now only donations with totalUsdStreamed greater than zero are counted. So, streams with small amount of amountStreamed may not be counted (We only count based on totalUsdStreamed greater than zero not amountStreamed greater than zero)

**Comment by maryjaf** [comment]
Based on the records of one time and recurring donation tables , this user has 14 donation but in overview tab, number of donation is shown 13 

https://staging.giveth.io/user/0xf577Ae8b97d839B9c0522A620299DC08792c738c?tab=donations

and number of donations in this  project  page https://staging.giveth.io/project/testhomepage  is shown 18 
but based on records in all donation table it has 12 one time donation more than 0$ and based on records in recurring donation table it has 5  recurring donation more than 0 $ and the total of them should be 17 donations
@RamRamez let me know if needed check this together

**Comment by RamRamez** [comment]
> * [x]  Based on the records of one time and recurring donation tables , this user has 14 donation but in overview tab, number of donation is shown 13
> 
> https://staging.giveth.io/user/0xf577Ae8b97d839B9c0522A620299DC08792c738c?tab=donations
> 
> * [x]  Number of donations in this  project  page https://staging.giveth.io/project/testhomepage  is shown 18
> 
> but based on records in all donation table it has 12 one time donation more than 0$ and based on records in recurring donation table it has 5 recurring donation more than 0 $ and the total of them should be 17 donations @RamRamez let me know if needed check this together

In user view, we only show amount donated and we don't show USD value there, and we don't count zero USD donations like 0.0000001 ETH

In project view, we count all normal donations, because we don't have zero donations. But for SF, we only count greater than zero donations.

I checked them both and everything is ok
